rskj-core/src/test/java/co/rsk/peg/RegisterBtcTransactionIT.java

```diff
@@ -26,131 +25,153 @@
-
         bridgeSupport.save();
-        track.commit();
 
         // Assert
         Optional<Long> heightIfBtcTxHashIsAlreadyProcessed = bridgeStorageProvider.getHeightIfBtcTxhashIsAlreadyProcessed(bitcoinTransaction.getHash());
-
         assertTrue(heightIfBtcTxHashIsAlreadyProcessed.isPresent());
-        assertEquals(rskExecutionBlock.getNumber(), heightIfBtcTxHashIsAlreadyProcessed.get());
+        assertEquals(RSK_EXECUTION_BLOCK_NUMBER, heightIfBtcTxHashIsAlreadyProcessed.get());
+
+        int outputIndex = 0;
+        TransactionOutput output = bitcoinTransaction.getOutput(outputIndex);
+        List<UTXO> expectedFederationUtxos = Collections.singletonList(utxoOf(bitcoinTransaction, output));
         assertEquals(expectedFederationUtxos, federationSupport.getActiveFederationBtcUTXOs());
-        assertEquals(expectedReceiverBalance, repository.getBalance(receiver));
 
-        verify(bridgeEventLogger, times(1)).logPeginBtc(receiver, bitcoinTransaction, btcTransferred, 0);
-    }
+        co.rsk.core.Coin expectedReceiverBalance = co.rsk.core.Coin.fromBitcoin(output.getValue());
+        assertEquals(expectedReceiverBalance, repository.getBalance(rskReceiver));
 
-    private static UTXO utxoOf(BtcTransaction bitcoinTransaction, TransactionOutput output) {
-        return new UTXO(
-                bitcoinTransaction.getHash(),
-                output.getIndex(),
-                output.getValue(),
-                0,
-                bitcoinTransaction.isCoinBase(),
-                output.getScriptPubKey()
-        );
+        assertLogPegInBtc();
     }
 
-    private static FederationSupport getFederationSupport(FederationStorageProvider federationStorageProvider, ActivationConfig.ForBlock activationConfig, FederationConstants federationConstants) {
-        return FederationSupportBuilder.builder()
-                .withFederationConstants(federationConstants)
-                .withFederationStorageProvider(federationStorageProvider)
-                .withActivations(activationConfig)
-                .build();
-    }
+    @Test
+    void registerBtcTransaction_forARepeatedLegacyBtcTransaction_shouldNotPerformAnyChange() throws Exception {
+        // Arrange
+        bridgeSupport.registerBtcTransaction(rskTx, bitcoinTransaction.bitcoinSerialize(), btcBlockWithPmtHeight, pmtWithTransactions.bitcoinSerialize());
+        bridgeSupport.save();
 
-    private FederationStorageProvider getFederationStorageProvider(Repository track, Federation federation) {
-        FederationStorageProvider federationStorageProvider = createFederationStorageProvider(track);
-        federationStorageProvider.setNewFederation(federation);
-        return federationStorageProvider;
-    }
+        co.rsk.core.Coin expectedReceiverBalance = repository.getBalance(rskReceiver);
+        List<UTXO> expectedFederationUTXOs = federationSupport.getActiveFederationBtcUTXOs();
```

Comment:
i just noticed that here you are pointing the reference to the utxos, so you end up asserting
`federationSupport.getActiveFederationBtcUTXOs().equals(federationSupport.getActiveFederationBtcUTXOs())`